homeassistant/components/webostv/media_player.py

```diff
@@ -65,127 +57,63 @@
-        if host in _CONFIGURING:
-            # Try to pair.
-            try:
-                client.register()
-            except PyLGTVPairException:
-                _LOGGER.warning("Connected to LG webOS TV %s but not paired", host)
-                return
-            except (OSError, ConnectionClosed, asyncio.TimeoutError):
-                _LOGGER.error("Unable to connect to host %s", host)
-                return
-        else:
-            # Not registered, request configuration.
-            _LOGGER.warning("LG webOS TV %s needs to be paired", host)
-            request_configuration(
-                host,
-                name,
-                customize,
-                config,
-                timeout,
-                hass,
-                add_entities,
-                turn_on_action,
+    @wraps(func)
+    async def wrapper(obj, *args, **kwargs):
+        """Wrap all command methods."""
+        try:
+            await func(obj, *args, **kwargs)
+        except (
+            asyncio.TimeoutError,
+            asyncio.CancelledError,
+            PyLGTVCmdException,
+        ) as exc:
+            # If TV is off, we expect calls to fail.
+            if obj.state == STATE_OFF:
+                log_function = _LOGGER.info
+            else:
+                log_function = _LOGGER.error
+            log_function(
+                "Error calling %s on entity %s: %r", func.__name__, obj.entity_id, exc
             )
-            return
-
-    # If we came here and configuring this host, mark as done.
-    if client.is_registered() and host in _CONFIGURING:
-        request_id = _CONFIGURING.pop(host)
-        configurator = hass.components.configurator
-        configurator.request_done(request_id)
-
-    add_entities(
-        [LgWebOSDevice(host, name, customize, config, timeout, hass, turn_on_action)],
-        True,
-    )
-
-
-def request_configuration(
-    host, name, customize, config, timeout, hass, add_entities, turn_on_action
-):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
-
-    # We got an error if this method is called while we are configuring
-    if host in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING[host], "Failed to pair, please try again."
-        )
-        return
-
-    def lgtv_configuration_callback(data):
-        """Handle actions when configuration callback is called."""
-        setup_tv(
-            host, name, customize, config, timeout, hass, add_entities, turn_on_action
-        )
 
-    _CONFIGURING[host] = configurator.request_config(
-        name,
-        lgtv_configuration_callback,
-        description="Click start and accept the pairing request on your TV.",
-        description_image="/static/images/config_webos.png",
-        submit_caption="Start pairing request",
-    )
+    return wrapper
 
 
-class LgWebOSDevice(MediaPlayerDevice):
+class LgWebOSMediaPlayerDevice(MediaPlayerDevice):
```

Comment:
Please name the class `LgWebOSMediaPlayerEntity` instead. We prefer to name entity classes entity now.